cibuildwheel/util.py

```diff
@@ -61,9 +63,12 @@ def __init__(self, build_config: str, skip_config: str):
         self.skip_patterns = skip_config.split()
 
     def __call__(self, build_id: str) -> bool:
-        def match_any(patterns: List[str]) -> bool:
-            return any(fnmatch(build_id, pattern) for pattern in patterns)
-        return match_any(self.build_patterns) and not match_any(self.skip_patterns)
+        build_patterns = itertools.chain.from_iterable(bracex.expand(p) for p in self.build_patterns)
+        skip_patterns = itertools.chain.from_iterable(bracex.expand(p) for p in self.skip_patterns)
+
+        build: bool = any(fnmatch.fnmatch(build_id, pat) for pat in build_patterns)
```

Comment:
Why not keep the same as before, where `match_any` contains the code duplication? (Which also have the nice property of short-circuiting, though it's not like we should be careful about performance here).

Reply:
I'm fine duplicating code twice, especially right next to each other, instead of defining a function in a function that is not a closure. I like the DRY-2 rule, where you are allowed to repeat yourself up to a maximum of twice if it's ugly to combine them into one. Pulling this out into a local function would be okay, but I think I liked the logic here inline better.

Reply:
I'm easy either way